lxd/devices.go

```diff
@@ -644,6 +645,11 @@ func deviceEventListener(stateFunc func() *state.State) {
 				continue
 			}
 
+			// Only auto pin VM CPUs when pin strategy is set to "auto"
```

Comment:
Why didnt you add this logic into ` TaskSchedulerTrigger` to avoid having to put another value into the event?

Reply:
Oh this logic is wrong anyway, where ever it is :)

It needs to be in deviceTaskBalance (which already has access to the instance's expanded config so it doesn't need to be passed via the event), because if `limits.cpu` is set to a static pinning then we should apply that irrespective of the value of the pin strategy setting.

Reply:
We'll also want some validation when the settings are applied directly to the instance's local config to ensure that the pin strategy isnt set to `auto` when using a non-integer `limits.cpu` setting.